tests.py

```diff
@@ -472,68 +472,111 @@ def test_csv_download_in__csv_listen_out(self):
 
-
     def test_bin_download_in__csv_listen_out(self):
-        global LOGGING_LEVEL
-        global TESTING_OUT_FILE_CSV, TESTING_IN_FILE_CSV, TESTING_IN_FILE_BIN
-        logging.basicConfig(format='%(asctime)s %(message)s', level=LOGGING_LEVEL)
+        host="localhost"
+        port=5015
+
+        a_pipeline = Pipeline()
+        a_file_source = FileSource( filename=TESTING_IN_FILE_CSV,
+                                    pipeline=a_pipeline)
+        a_pipeline.add_processing_step(ListenSink(  max_receivers=5,
+                                                    host=host,
+                                                    port=port))
+        a_file_source.start()
+        a_pipeline.start()
+        b_pipeline = Pipeline()
+        b_download_source = DownloadSource( host=host,
+                                            port=port,
+                                            pipeline=b_pipeline)
+        b_pipeline.add_processing_step(FileSink(filename=TESTING_OUT_FILE_BIN,
+                                                data_format=BINARY_DATA_FORMAT_IDENTIFIER))
+        b_pipeline.start()
+        b_download_source.start()
+        time.sleep(self.DEFAULT_SLEEPING_DURATION)
+        a_file_source.stop()
+        a_pipeline.stop()
+        b_download_source.stop()
+        b_pipeline.stop()
 
-        remove_file(TESTING_OUT_FILE_CSV)
-        remove_file(TESTING_OUT_FILE_CSV_2)
+        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_BIN,TESTING_OUT_FILE_BIN))
 
+    def test_csv_download_in__bin_listen_out(self):
         host="localhost"
-        port=5013
+        port=5016
 
         a_pipeline = Pipeline()
         a_file_source = FileSource( filename=TESTING_IN_FILE_BIN,
                                     pipeline=a_pipeline)
-        a_pipeline.add_processing_step(ListenSink(max_receivers=5,host=host,port=port))
+        a_pipeline.add_processing_step(ListenSink(  max_receivers=5,
+                                                    host=host,
+                                                    port=port))
         a_file_source.start()
         a_pipeline.start()
+        b_pipeline = Pipeline()
+        b_download_source = DownloadSource( host=host,
+                                            port=port,
+                                            pipeline=b_pipeline)
+        b_pipeline.add_processing_step(FileSink(filename=TESTING_OUT_FILE_CSV,
+                                                data_format=CSV_DATA_FORMAT_IDENTIFIER))
+        b_pipeline.start()
+        b_download_source.start()
+        time.sleep(self.DEFAULT_SLEEPING_DURATION)
+        a_file_source.stop()
+        a_pipeline.stop()
+        b_download_source.stop()
+        b_pipeline.stop()
 
-        time.sleep(5)
+        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_CSV,TESTING_OUT_FILE_CSV))
 
+    def test_bin_download_in__csv_listen_out(self):
+        host="localhost"
+        port=5017
```

Comment:
Tests should also not depend on fixed ports... There is a mechanism to let the operating system pick a free port for you, I think by using port 0. Then you must check the selected port after opening the socket.